test/Dialect/FIRRTL/extract-instances.mlir

```diff
@@ -0,0 +1,129 @@
+// RUN: circt-opt --firrtl-extract-instances --verify-diagnostics %s | FileCheck %s
+
+// Tests extracted from:
+// - test/scala/firrtl/ExtractBlackBoxes.scala
+
+//===----------------------------------------------------------------------===//
+// ExtractBlackBoxes Simple
+//===----------------------------------------------------------------------===//
+
+// CHECK: firrtl.circuit "ExtractBlackBoxesSimple"
+firrtl.circuit "ExtractBlackBoxesSimple" attributes {annotations = [{class = "firrtl.transforms.BlackBoxTargetDirAnno", targetDir = "BlackBoxes"}]} {
+  // CHECK-LABEL: firrtl.extmodule private @MyBlackBox
+  firrtl.extmodule private @MyBlackBox(in in: !firrtl.uint<8>, out out: !firrtl.uint<8>) attributes {annotations = [{class = "sifive.enterprise.firrtl.ExtractBlackBoxAnnotation", filename = "BlackBoxes.txt", prefix = "bb"}], defname = "MyBlackBox"}
+  // CHECK-LABEL: firrtl.module private @BBWrapper
+  // CHECK-SAME: out %bb_0_in: !firrtl.uint<8>
+  // CHECK-SAME: in %bb_0_out: !firrtl.uint<8>
+  firrtl.module private @BBWrapper(in %clock: !firrtl.clock, in %reset: !firrtl.reset, out %io: !firrtl.bundle<in flip: uint<8>, out: uint<8>>) {
+    %0 = firrtl.subfield %io(0) : (!firrtl.bundle<in flip: uint<8>, out: uint<8>>) -> !firrtl.uint<8>
+    %1 = firrtl.subfield %io(1) : (!firrtl.bundle<in flip: uint<8>, out: uint<8>>) -> !firrtl.uint<8>
+    // CHECK-NOT: firrtl.instance bb @MyBlackBox
+    %bb_in, %bb_out = firrtl.instance bb  @MyBlackBox(in in: !firrtl.uint<8>, out out: !firrtl.uint<8>)
+    %invalid_ui8 = firrtl.invalidvalue : !firrtl.uint<8>
+    firrtl.strictconnect %bb_in, %invalid_ui8 : !firrtl.uint<8>
+    // CHECK: firrtl.connect %1, %bb_0_out
+    // CHECK: firrtl.connect %bb_0_in, %0
+    firrtl.connect %1, %bb_out : !firrtl.uint<8>, !firrtl.uint<8>
+    firrtl.connect %bb_in, %0 : !firrtl.uint<8>, !firrtl.uint<8>
+  }
+  // CHECK-LABEL: firrtl.module private @DUTModule
+  // CHECK-SAME: out %bb_0_in: !firrtl.uint<8>
+  // CHECK-SAME: in %bb_0_out: !firrtl.uint<8>
+  firrtl.module private @DUTModule(in %clock: !firrtl.clock, in %reset: !firrtl.reset, out %io: !firrtl.bundle<in flip: uint<8>, out: uint<8>>) attributes {annotations = [{class = "sifive.enterprise.firrtl.MarkDUTAnnotation"}]} {
```

Comment:
Is this supposed to put the module into the DUT or outside the DUT?  I thought it was put it into the DUT.

Reply:
Outside the DUT according to the SFC unit tests. See https://github.com/llvm/circt/pull/3017#discussion_r863993417

Reply:
Update: *in* the DUT if the `dest` field in the anno is set; otherwise *outside* the DUT :see_no_evil: